components/utilities/ymodem/ymodem.c

```diff
@@ -7,12 +7,17 @@
  * Change Logs:
  * Date           Author       Notes
  * 2013-04-14     Grissiom     initial implementation
+ * 2019-12-09     Steven Liu   add YMODEM send protocol
  */
 
 #include <rthw.h>
 #include "ymodem.h"
 
-static const rt_uint16_t ccitt_table[256] = {
+#define CRC_TABLE
```

Comment:
If you want to make CRC table as an option, you can use YMODEM_USING_CRC_TABLE and place it in KConfig.